src/main/java/org/beanpod/switchboard/dao/StreamDaoImpl.java

```diff
@@ -31,20 +34,30 @@ public StreamDTO getStreamById(Long id) {
     return mapper.toDto(streamEntity);
   }
 
-  public void createStream(CreateStreamRequest createStreamRequest) {
-    InputChannelDTO inputChannelDTO =
-        channelService.getInputChannelById(createStreamRequest.getInputChannelId());
-    OutputChannelDTO outputChannelDTO =
-        channelService.getOutputChannelById(createStreamRequest.getOutputChannelId());
-    StreamDTO streamDto =
-        StreamDTO.builder().inputChannel(inputChannelDTO).outputChannel(outputChannelDTO).build();
-    StreamEntity streamEntity = mapper.toEntity(streamDto);
+  public StreamDTO createStream(CreateStreamRequest createStreamRequest) {
     if (streamRepository.existsDuplicate(
         createStreamRequest.getInputChannelId(), createStreamRequest.getOutputChannelId())) {
       throw new ExceptionType.StreamAlreadyExistsException(
           createStreamRequest.getInputChannelId(), createStreamRequest.getOutputChannelId());
     }
-    streamRepository.save(streamEntity);
+
+    InputChannelDTO inputChannelDto =
+        channelService.getInputChannelById(createStreamRequest.getInputChannelId());
+    OutputChannelDTO outputChannelDto =
+        channelService.getOutputChannelById(createStreamRequest.getOutputChannelId());
+    StreamDTO.StreamDTOBuilder streamDtoBuilder =
+        StreamDTO.builder().inputChannel(inputChannelDto).outputChannel(outputChannelDto);
+
+    if (onSamePrivateNetwork(inputChannelDto, outputChannelDto)
+        || onLocalNetwork(inputChannelDto, outputChannelDto)) {
+      streamDtoBuilder.mode(StreamModeDTO.SRT);
+    } else {
+      streamDtoBuilder.mode(StreamModeDTO.SRT_RENDEZVOUS);
+    }
```

Comment:
Should we extract this into its own `selectMode` method or is that overkill?

Reply:
It's extracted in the `StreamServiceImpl` now.